superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -259,6 +266,12 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
       }
     } else if (db) {
       // Create
+      if (update.encrypted_extra) {
+        // wrap encrypted_extra in credentials_info
+        update.encrypted_extra = JSON.stringify({
+          credentials_info: JSON.parse(update.encrypted_extra),
```

Comment:
you can do this part in the reducer and keep the db model as close to the schema that the api is going to expect.

Reply:
This always breaks in the reducer since the user is typing, i rather have do the transform on save vs. having to put it in a try/catch